okhttp/src/main/kotlin/okhttp3/internal/http/CallServerInterceptor.kt

```diff
@@ -58,9 +59,15 @@ class CallServerInterceptor(private val forWebSocket: Boolean) : Interceptor {
             requestBody.writeTo(bufferedRequestBody)
           } else {
             // Write the request body if the "Expect: 100-continue" expectation was met.
-            val bufferedRequestBody = exchange.createRequestBody(request, false).buffer()
-            requestBody.writeTo(bufferedRequestBody)
-            bufferedRequestBody.close()
+            val requestBodySink = exchange.createRequestBody(request, false)
+            val bufferedRequestBody = requestBodySink.buffer()
+            try {
+              requestBody.writeTo(bufferedRequestBody)
+              bufferedRequestBody.close()
+            } finally {
+              requestInProgress = requestBodySink.inProgress();
+              // TODO move bufferedRequestBody.close() here?
```

Comment:
closing the RequestBodySink here fixes ServerTruncatesRequestTest.readResponseWhenPartialRequest(). And it does make sense to always close the RequestBodySink right?

Reply:
Or put it inside a `.use {}` block?

Reply:
Would be nice, but `close()` completes the request body and currently this affects `inProgress()`. So i think `inProgress()` must be called before `close()` when the transfer fails.

But if i reimplement it as `didReceiveBytes()`, i can use `.use{}` :+1: